dynawo/sources/Models/Modelica/Dynawo/Electrical/Sources/InjectorIDQ.mo

```diff
@@ -16,68 +16,67 @@ model InjectorIDQ "Injector controlled by d and q current components idPu and iq
     Placement(visible = true, transformation(extent = {{0, 0}, {0, 0}}, rotation = 0), iconTransformation(origin = {115, -79}, extent = {{-15, -15}, {15, 15}}, rotation = 0)));
 
-  parameter Types.ApparentPowerModule SNom "Nominal apparent power in MVA";
-
-  // Inputs: d-q axis pu variables (base UNom, SNom)
+  // Inputs: d-q axis pu variables (base UNom, SNom) and rotor angle
   Modelica.Blocks.Interfaces.RealInput idPu(start = Id0Pu) "Injected d-axis current in pu (base SNom, UNom)" annotation(
     Placement(visible = true, transformation(extent = {{0, 0}, {0, 0}}, rotation = 0), iconTransformation(origin = {-115, 61}, extent = {{-15, -15}, {15, 15}}, rotation = 0)));
   Modelica.Blocks.Interfaces.RealInput iqPu(start = Iq0Pu) "Injected q-axis current in pu (base SNom, UNom)" annotation(
     Placement(visible = true, transformation(extent = {{0, 0}, {0, 0}}, rotation = 0), iconTransformation(origin = {-115, -41}, extent = {{-15, -15}, {15, 15}}, rotation = 0)));
+  Modelica.Blocks.Interfaces.RealInput UPhase(start = UPhase0) "Rotor angle: angle between machine rotor frame and port phasor frame in rad" annotation(
+    Placement(visible = true, transformation(extent = {{0, 0}, {0, 0}}, rotation = 0), iconTransformation(origin = {0, 115}, extent = {{-15, -15}, {15, 15}}, rotation = -90)));
 
   // Outputs:
-  Modelica.Blocks.Interfaces.RealOutput UPu(start = U0Pu) "Magnitude voltage at inverter terminal in pu (base UNom)" annotation(
-    Placement(visible = true, transformation(extent = {{0, 0}, {0, 0}}, rotation = 0), iconTransformation(origin = {115, 81}, extent = {{-15, -15}, {15, 15}}, rotation = 0)));
-  Modelica.Blocks.Interfaces.RealOutput QInjPuSn(start = -Q0Pu * SystemBase.SnRef /SNom) "Injected reactive power in pu (base SNom)" annotation(
-    Placement(visible = true, transformation(extent = {{0, 0}, {0, 0}}, rotation = 0), iconTransformation(origin = {115, 5}, extent = {{-15, -15}, {15, 15}}, rotation = 0)));
+  Modelica.Blocks.Interfaces.RealOutput PInjPu(start = -P0Pu) "Injected active power in pu (base SnRef)";
   Modelica.Blocks.Interfaces.RealOutput PInjPuSn(start = -P0Pu * SystemBase.SnRef /SNom) "Injected active power in pu (base SNom)" annotation(
     Placement(visible = true, transformation(extent = {{0, 0}, {0, 0}}, rotation = 0), iconTransformation(origin = {115, 43}, extent = {{-15, -15}, {15, 15}}, rotation = 0)));
   Modelica.Blocks.Interfaces.RealOutput QInjPu(start = -Q0Pu) "Injected reactive power in pu (base SnRef)";
-  Modelica.Blocks.Interfaces.RealOutput PInjPu(start = -P0Pu) "Injected active power in pu (base SnRef)";
+  Modelica.Blocks.Interfaces.RealOutput QInjPuSn(start = -Q0Pu * SystemBase.SnRef /SNom) "Injected reactive power in pu (base SNom)" annotation(
+    Placement(visible = true, transformation(extent = {{0, 0}, {0, 0}}, rotation = 0), iconTransformation(origin = {115, 5}, extent = {{-15, -15}, {15, 15}}, rotation = 0)));
```

Comment:
Thanks for putting these outputs in the alphabetical order. Moreover, could you :

- Add the convention  for all four outputs : "Injected active power in pu (base SnRef) (generator convention)"
- Insert a space in '/SNom', which would become '/ SNom'